GVFS/GVFS.UnitTests/Common/Database/PlaceholderTableTests.cs

```diff
@@ -0,0 +1,668 @@
+﻿using GVFS.Common.Database;
+using GVFS.Tests.Should;
+using GVFS.UnitTests.Category;
+using Moq;
+using NUnit.Framework;
+using System;
+using System.Collections.Generic;
+using System.Data;
+using System.Linq;
+
+namespace GVFS.UnitTests.Common.Database
+{
+    [TestFixture]
+    public class PlaceholderTableTests
+    {
+        private const string DefaultExceptionMessage = "Somethind bad.";
+        private const string DefaultPath = "test";
+        private const byte PathTypeFile = 0;
+        private const byte PathTypePartialFolder = 1;
+        private const byte PathTypeExpandedFolder = 2;
+        private const byte PathTypePossibleTombstoneFolder = 3;
+        private const string DefaultSha = "1234567890123456789012345678901234567890";
+
+        [TestCase]
+        public void ConstructorTest()
+        {
+            Mock<IGVFSConnectionPool> mockConnectionPool = new Mock<IGVFSConnectionPool>(MockBehavior.Strict);
+            PlaceholderTable placeholders = new PlaceholderTable(mockConnectionPool.Object);
+            mockConnectionPool.VerifyAll();
+        }
+
+        [TestCase]
+        public void CreateTableTest()
+        {
+            Mock<IDbCommand> mockCommand = new Mock<IDbCommand>(MockBehavior.Strict);
+            mockCommand.SetupSet(x => x.CommandText = "CREATE TABLE IF NOT EXISTS [Placeholder] (path TEXT PRIMARY KEY, pathType TINYINT NOT NULL, sha char(40) ) WITHOUT ROWID;");
+            mockCommand.Setup(x => x.ExecuteNonQuery()).Returns(1);
+            mockCommand.Setup(x => x.Dispose());
+
+            Mock<IDbConnection> mockConnection = new Mock<IDbConnection>(MockBehavior.Strict);
+            mockConnection.Setup(x => x.CreateCommand()).Returns(mockCommand.Object);
+
+            PlaceholderTable.CreateTable(mockConnection.Object);
+            mockCommand.VerifyAll();
+            mockConnection.VerifyAll();
+        }
+
+        [TestCase]
+        [Category(CategoryConstants.ExceptionExpected)]
+        public void CreateTableThrowsExceptionNotWrappedInGVFSDatabaseException()

```

Comment:
Oh it looks like it is expected the Create doesn't throw `GVFSDatabaseException`